### PR TITLE
Refactor earn memberships component

### PR DESCRIPTION
**Pull request by edanzer**
## Proposed Changes

Converts the Earn > MembershipsSection from class to function component.

## Testing Instructions

There should be no change in functionality before/after this PR. 

1) Load this branch and go to the Earn > Memberships page at `http://calypso.localhost:3000/earn/payments/YOURDOMAIN`.
   - The component being refactored manages the content on this page. So we'll want to test all the functionality.
   - You can always compare what you see against production by opening `https://wordpress.com/earn/payments/YOURDOMAIN` in a separate tab and going back and forth.

2) Test without Stripe connection: Before you connect Stripe, you should see the 'Accept payments on your website' intro/onboarding card, along with a button to connect Stripe. Confirm that loads as expected. Click the button to connect Stripe, connect it, and return. 

without stripe

3) Test with Stripe connection. After you are connected to Stripe, the page will show a number of different sections. Go section by section and just test that everything looks and works as expected. 
   - The most complicated thing to test, but worth doing, is to actually buy a product as a customer. You'll need to have Stripe connected. Create a paid newsletter plan with small one-time payment of $1. Go the front end of post with the subscribe block active, enter a different email, click subscribe, chose a paid plan, and pay. Return to this screen, and confirm that your new customer shows in the customer list, and that your new revenue shows in the stats section at the top. 

with stripe

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/earn/memberships/index.jsx [hunk 1]
```diff
@@ -643,117 +637,83 @@ class MembershipsSection extends Component {
...
-		source,
-	};
-};
-
-export default connect( mapStateToProps, {
-	recordTracksEvent,
-	requestSubscribers,
-	requestDisconnectSiteStripeAccount,
-	requestSubscriptionStop,
-} )( localize( withLocalizedMoment( MembershipsSection ) ) );
```
Comment (edanzer) [inline client/my-sites/earn/memberships/index.jsx:759]: One small thing that isn't matched 1-for-1 in this PR is withLocalizedMoment. I played with this a bit, but it was throwing errors. So I'm just importing moment directly and using that. Using moment directly is the more common use in the calypso code base - there are 100+ files that import and use moment directly. But if we think it's a big deal to keep this, I can continue to work on that piece.

**Review by simison — APPROVED** [review]
Tested using stripe connected, stripe not connected, and a Jetpack site. Tested disconnecting Stripe.

All looks good and seems to function as before.

File: client/my-sites/earn/memberships/index.jsx (hunk 1 above)
Comment (simison) [reply client/my-sites/earn/memberships/index.jsx:759]: [There's also `useLocalizedMoment`](https://github.com/Automattic/wp-calypso/blob/4413cf0ea22b368ba9f5d2da12b6ed17bb025bba/client/components/localized-moment/README.md#uselocalizedmoment) — let's use that instead. OK to do in a follow up.

Comment (edanzer) [reply client/my-sites/earn/memberships/index.jsx:759]: Good find! I'll update that in an immediate follow up.

**Pull request merged**